src/tpm.rs

```diff
@@ -90,37 +112,93 @@ impl Tpm {
-            reg.c4v.write(pwm_trigger as u32);
+            //Set CnSC to 0 to allow for writing.
+            //Write to CnSC
+            //Write to CnV (trigger value for PWM). 
+            match channel_select {
+                ChannelSel::Ch0 => {
+                    reg.c0sc.write(0);
+                    reg.c0sc.write(channel_set as u32);
+                    reg.c0v.write(pwm_trigger as u32);
+                },
+                ChannelSel::Ch1 => {
+                    reg.c1sc.write(0);
+                    reg.c1sc.write(channel_set as u32);
+                    reg.c1v.write(pwm_trigger as u32);
+
+                },
+                ChannelSel::Ch2 => {
+                    reg.c2sc.write(0);
+                    reg.c2sc.write(channel_set as u32);
+                    reg.c2v.write(pwm_trigger as u32);
+                },
+                ChannelSel::Ch3 => {
+                    reg.c3sc.write(0);
+                    reg.c3sc.write(channel_set as u32);
+                    reg.c3v.write(pwm_trigger as u32);
+                },
+                ChannelSel::Ch4 => {
+                    reg.c4sc.write(0);
+                    reg.c4sc.write(channel_set as u32);
+                    reg.c4v.write(pwm_trigger as u32);
+                },
+                ChannelSel::Ch5 => {
+                    reg.c5sc.write(0);
+                    reg.c5sc.write(channel_set as u32);
+                    reg.c5v.write(pwm_trigger as u32);
+
+                },
+            }
 
             // Enable the TPM.
             reg.sc.modify(|mut sc| {
                 sc.set_bits(3..5, cmod as u32);
                 sc
             });
 
-            Tpm {
-                _reg: reg,
-            }
+            //Demo code. Delays were arbitrary times to allow the actuator to move fully.
+            //This also demonstrates that writing to CnV after the setup does what the datasheet implies.
+            //In this case, cmod is on w/ EPWM selected: write after counter goes from MOD to 0. 
+            /*asm::delay(400_000_000);
+
+            reg.c4v.write(Positions::FullyRetracted as u32);
+
+            asm::delay(200_000_000);
+
+            reg.c4v.write(Positions::Middle as u32);
+
+            asm::delay(200_000_000);
+
+            reg.c4v.write(Positions::FullyActuated as u32);
+
+            asm::delay(200_000_000);
+
+            reg.c4v.write(Positions::FullyRetracted as u32);*/
+
+            Ok(Tpm {reg, _pin:pin})
     }
-}
 
+    //Attempt at writing to c4v through a setter.
+    //Writes work fine if put into the initialization block, but fail when using
+    //a helper function. In this case, it appears to get stuck in an infinite loop
+    //while trying ot write to the register. Writing to ANY tpm register in this fashion will
+    //cause the pin to produce identical sawtooth waveforms regardless of the register. 
+    pub fn set_trigger(&mut self) {
+        unsafe { self.reg.c4v.write(Positions::FullyRetracted as u32); }
```

Comment:
I know this is for testing, but I want to leave a comment to mark this for the future.

Don't assume what channel was set up.